### ISSUE TITLE
Place cursor in email text field for sign in

### ISSUE THREAD

**Issue by ariel-phet** [phetsims/website#214]
@aaronsamuel137 - this is not a high priority enhancement, but it would be a nice touch

When you click the "Sign in" button, my instinct is to begin immediately typing my email address, but I first need to click the email box. It seems the standard for most sign in (such as gmail and such) is to have the cursor already active in the appropriate text entry field. It seems the ideal interaction would be that when you click our sign in button that the email text entry field is preselected.

**Comment by aaronsamuel137** [comment]
This behavior actually was implemented at some point, but one of my changes must have accidentally broken it.

**Comment by aaronsamuel137** [comment]
This should be working on the next deploy.

**Comment by amanda-phet** [comment]
If the cursor is automatically there, we just need to be sure the "Email address" text is still visible before typing.

**Comment by aaronsamuel137** [comment]
The text should still be visible even with the cursor there. This should be working now. Assigning to @ariel-phet for review.

**Comment by ariel-phet** [comment]
Awesome! Looks great on FF and Chrome on windows, but text is not visible on IE

Can you check into that?

**Comment by aaronsamuel137** [comment]
So far I am not seeing an easy way to do fix this for IE. Looks like @amanda-phet was right to have some concern about this problem. Leave it to IE to be inconsistent with other browsers.

However, this issue brings up a whole other concern around our use of placeholders in the UX of our site. The main concern seems to be accessibility, and the HTML5 spec strongly warns against using placeholders instead of labels:

> Use of the placeholder attribute as a replacement for a label can reduce the accessibility and usability of the control for a range of users including older users and users with cognitive, mobility, fine motor skill or vision impairments. While the hint given by the control's label is shown at all times, the short hint given in the placeholder attribute is only shown before the user enters a value. Furthermore, placeholder text may be mistaken for a pre-filled value, and as commonly implemented the default color of the placeholder text provides insufficient contrast and the lack of a separate visible label reduces the size of the hit region available for setting focus on the control.

From http://www.w3.org/html/wg/drafts/html/master/semantics.html#the-placeholder-attribute

**Comment by aaronsamuel137** [comment]
There are probably some ways to work around the IE limitation if that seems like the way we want to go, but before I spend a bunch of time on that it would be good to discuss the accessibility points and decide if adding labels here might be a good idea.

**Comment by amanda-phet** [comment]
I think the accessibility points you mention about very good- and after we made some of these design choices I had read about them myself. I think that for a "sign in" dialog, it's standard to have email and password as the two entry boxes, so I am not too concerned about the usability there. However, it is something we should not use in the future, and we may want to even modify the registration page where placeholders are used as labels. 

If you want to change the sign in dialog to use labels (consistent with the others found on the activity submission page) I think that's fine, and then we can have the cursor in the email text field! @ariel-phet what do you think?

**Comment by zepumph** [comment]
I don't mind changing the register panel to labels, I'll use the activity submission page as a base.

**Comment by aaronsamuel137** [comment]
@zepumph, I would check with @ariel-phet before we go ahead with this - there might be some redesign work that needs to happen first.

**Comment by aaronsamuel137** [comment]
@zepumph - a good thing to look into here would be a workaround so that the words "Email address" appear in IE in the log in form even when the cursor is there - like it does in other browsers. It'll probably take some CSS and JS hacks to accomplish this.

**Comment by zepumph** [comment]
Assigning to @ariel-phet  to see about a design changes in the register panel.

**Comment by ariel-phet** [comment]
Let me talk to @amanda-phet

**Comment by zepumph** [comment]
As discussed in the 7/23 status meeting. Regarding the redesign of the register panel, there should be a note that all fields are required, or perhaps a * next to every field.

**Comment by amanda-phet** [comment]
@zepumph I think this is the wrong issue.

**Comment by zepumph** [comment]
@amanda-phet This was the one I was thinking of because there is talk of a redesign for the register page, I can put it in a new issue also though.

**Comment by ariel-phet** [comment]
@amanda-phet - this point still seems to warrant some discussion. Perhaps at an upcoming website meeting?

**Comment by amanda-phet** [comment]
Is there much to discuss? I thought we had decided that it is a better practice to not put the label inside of the entry box, and we are going to change that on the registration page. @aaronsamuel137 said above that he does not need a new design:
>I'll use the activity submission page as a base. 

I think @zepumph can move forward with this on the registration page. If you want to discuss the sign-in dialog, though, that is fine with me. 

I commented that this was the wrong issue because of the title, but I see that the original issue had evolved.

**Comment by ariel-phet** [comment]
@amanda-phet I meant the sign in dialog (since it still has odd behavior on IE)

**Comment by zepumph** [comment]
There is a version of the label register panel on phet-dev right now, @amanda-phet whatcha think?

**Comment by amanda-phet** [comment]
Looks good to me! Do you think we should keep the text inside the boxes, too? It is redundant, but I don't know if it's distracting or helpful.

**Comment by zepumph** [comment]
Yeah, i'm not totally sure either. It just seems like it takes away from the page a little bit. What we can do is put a few hints in, like on the activity submission page, I'm not sure anything really warrants it, just a thought though. Dumb example : 
First Name
(in the text box) --> The name your mother gave you

I also just noticed I'm saying username instead of email, so I'll fix that as well.

**Comment by amanda-phet** [comment]
@zepumph Can you go ahead and make the same changes to the sign in dialog?

**Comment by zepumph** [comment]
The labels are added on the sign in pop up as well. On phet-dev, what do you think @amanda-phet.

**Comment by zepumph** [comment]
There are two different places this change is implemented, so you know. One is the top right on the home screen, and the other is whenever you have to login to access somewhere on the site, like myActivities. You can look at both. They look the same to me, but gosh knows I'm not an authority in that regard.

**Comment by amanda-phet** [comment]
There are a few other places (which clicking on an activity document or translate the website, for example) and they all look good to me.

**Comment by zepumph** [comment]
Closing issue unless anyone has anything else to report. It will be up on production the next deploy.

**Comment by aaronsamuel137** [comment]
Two more issues I notice here:

1. Labels on sign in popup are not bold like everywhere else
2. Edit profile page still shows labels for email and password even though the form inputs don't show up:

<img width="502" alt="screen shot 2015-08-04 at 1 50 07 pm" src="https://cloud.githubusercontent.com/assets/3193912/9070979/c16d867e-3aaf-11e5-9bf3-63c777d49764.png">

If you don't have time to fix this before you leave @zepumph, don't worry. I can take care of it. We just need to be sure to address if before the next deploy.

**Comment by zepumph** [comment]
Any other thoughts @aaronsamuel137?

**Comment by aaronsamuel137** [comment]
I just made one more commit for this, looking good now, thanks!

I am planning on doing a deploy tonight, any concerns with this?

**Comment by aaronsamuel137** [comment]
Deployed. Closing
